### PR TITLE
Add Python Executorch TaskProcessor for edge emulator

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Add TaskProcessor that use the executorch python bindings directly.

Later we can remove those CPP binary delegation TaskProcessor if these python bindings are good enough.

### Description

- Rename ETTaskProcessor -> ETCppTaskProcessor to reflect it delegates training to cpp binary
- Add ETTaskProcessor
- Add CIFAR10ETTaskProcessor
- Add XORETTaskProcessor
- Add corresponding config files

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/edge/emulator/cifar10_cpp.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "class_path": "et_cpp_task_processor.ETCppTaskProcessor",
+    "class_args": {"et_binary_path": "train_cifar10", "et_model_path": "cifar10.pte", "data_path": "/Users/yuantingh/Downloads/cifar-10-batches-bin/data_batch_1.bin"}
```
Comment (ZiyueXu77) [inline nvflare/edge/emulator/cifar10_cpp.json:3]: same here

File: nvflare/edge/emulator/cifar10_et_task_processor.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from et_task_processor import ETTaskProcessor
+from torch.utils.data import Dataset
+from torchvision import datasets, transforms
+
+
+class CIFAR10ETTaskProcessor(ETTaskProcessor):
+    def get_dataset(self, data_path: str) -> Dataset:
+        transform = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
```
Comment (ZiyueXu77) [inline nvflare/edge/emulator/cifar10_et_task_processor.py:8]: can the transform be performed on device (i.e. torchvision works on phone)?

**Review by ZiyueXu77 — COMMENTED** [review]
few questions

File: nvflare/edge/emulator/cifar10_et_task_processor.py (hunk 2 above)
Comment (YuanTingHsieh) [reply nvflare/edge/emulator/cifar10_et_task_processor.py:8]: torchvision lib does NOT work on phone, phone need to use other code to do the equivalent thing if they want to.

These code are just for our device emulator that runs in Mac/Linux

Comment (ZiyueXu77) [reply nvflare/edge/emulator/cifar10_et_task_processor.py:8]: For the demo, I think we can do pre-normalization, so that emulator devices and the actual training on your phone will have the same input (unless you already have the same normalization on your phone)

Comment (YuanTingHsieh) [reply nvflare/edge/emulator/cifar10_et_task_processor.py:8]: good point, let me be consistent across diff language

File: nvflare/edge/emulator/cifar10_cpp.json (hunk 1 above)
Comment (nvidianz) [reply nvflare/edge/emulator/cifar10_cpp.json:3]: Should change this to the new JSON format.

File: nvflare/edge/emulator/et_cpp_task_processor.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    with open(filename, "wb") as f:
+        f.write(binary_data)
+
+
+def run_training_with_timeout(
+    train_program: str, model_path: str, result_path: str, data_path: str = "", timeout_seconds: int = 300
+) -> int:
+    try:
+        command = [train_program, "--model_path", model_path, "--output_path", result_path, "--data_path", data_path]
+        process = subprocess.Popen(
```
Comment (nvidianz) [inline nvflare/edge/emulator/et_cpp_task_processor.py:50]: subprocess.run() is a simpler.

File: nvflare/edge/emulator/et_task_processor.py [hunk 4]
```diff
@@ -36,85 +41,78 @@
...
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            text=True,
-        )
+def tensor_dict_to_json(d):
+    j = {}
+    for k, v in d.items():
+        entry = {}
+        # TODO: this needs to be compatible with the "Controller" side logic
+        entry["data"] = v.cpu().numpy().tolist()
```
Comment (nvidianz) [inline nvflare/edge/emulator/et_task_processor.py:49]: base64(np.save()) is probably more compact..

**Review by nvidianz — APPROVED** [review]
Good enough for demo.
